src/EncompassApi/Extensions/HttpClientBuilderExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+
+namespace EncompassApi.Extensions
+{
+    public static class HttpClientBuilderExtensions
+    {
+        public static IHttpClientBuilder ConfigurPrimaryHttpClientHandler(this IHttpClientBuilder httpClientBuilder, IServiceProvider s)
+        {
+            var options = s.GetRequiredService<IOptions<Configuration.HttpClientOptions>>();
+            if (options != null)
+            {
+                var handler = new HttpClientHandler();
+
+                if (options.Value.CompressionOptions.DecompressionMethods != null && options.Value.CompressionOptions.EnableAutoDecompression)
+                {
+                    // DecompressionMethods.None should be removed
```

Comment:
Can you elaborate on this comment?

Reply:
None is the default for Net Core 2.0 +. What I am doing is setting a variable to None, so I can have the variable to use in the loop and build the list of decompressions with the operator |. There might be a more elegant way to do this. I am open to suggestions.